core/src/main/scala/scala/pickling/internal/DefaultPicklerRegistry.scala

```diff
@@ -14,10 +14,14 @@ final class DefaultPicklerRegistry(generator: RuntimePicklerGenerator) extends P
   type UnpicklerGenerator = FastTypeTag[_] => Unpickler[_]
   // TODO - We need to move the special encoding for runtime classes into here, rather than in magical traits.
 
-  private val picklerMap: mutable.Map[String, Pickler[_]] = new TrieMap[String, Pickler[_]]
-  private val picklerGenMap: mutable.Map[String, PicklerGenerator] = new TrieMap[String, PicklerGenerator]
-  private val unpicklerMap: mutable.Map[String, Unpickler[_]] = new TrieMap[String, Unpickler[_]]
-  private val unpicklerGenMap: mutable.Map[String, UnpicklerGenerator] = new TrieMap[String, UnpicklerGenerator]
+  import mutable.{Map => Mmap, Set => Mset}
+  private val picklerMap: Mmap[String, Pickler[_]] = new TrieMap[String, Pickler[_]]
+  private val picklerGenMap: Mmap[String, PicklerGenerator] = new TrieMap[String, PicklerGenerator]
+  private val unpicklerMap: Mmap[String, Unpickler[_]] = new TrieMap[String, Unpickler[_]]
+  private val unpicklerGenMap: Mmap[String, UnpicklerGenerator] = new TrieMap[String, UnpicklerGenerator]
+
+  private val shareNothingPicklers: Mset[String] = Mset.empty[String]
```

Comment:
Given that most users will be picking ONE share strategy for their entire protocol, I do think this is a pretty big overhead/burden on the API/users for such a small use case (specifically the test).